testdata/cheats/Sleep.t.sol

```diff
@@ -0,0 +1,51 @@
+// SPDX-License-Identifier: Unlicense
+pragma solidity 0.8.18;
+
+import "ds-test/test.sol";
+import "./Vm.sol";
+
+contract SleepTest is DSTest {
+    Vm constant vm = Vm(HEVM_ADDRESS);
+
+    function testSleep() public {
+        uint256 milliseconds = 1234;
+
+        string[] memory inputs = new string[](2);
+        inputs[0] = "date";
+        // OS X does not support precision more than 1 second
+        inputs[1] = "+%s000";
+
+        bytes memory res = vm.ffi(inputs);
+        uint256 start = vm.parseUint(string(res)); 
+
+        vm.sleep(milliseconds);
+
+        res = vm.ffi(inputs);
+        uint256 end = vm.parseUint(string(res)); 
+
+        // Limit precision to 1000 ms
+        assertGe(end - start, milliseconds / 1000 * 1000, "sleep failed");
+    }
+
+    /// forge-config: default.fuzz.runs = 10
+    function testSleepFuzzed(uint256 _milliseconds) public {
+        // Limit sleep time to 2 seconds to decrease test time
+        uint256 milliseconds = _milliseconds % 2000;
+
+        string[] memory inputs = new string[](2);
+        inputs[0] = "date";
```

Comment:
Any ideas how to measure sleep duration with millisecond precision cross-platform? 

One idea is to implement a custom vm cheatcode for system time...

Reply:
Ah right, properly introducing tests for this might not be so easy, and they might be a bit flaky honestly due to cross-platform issues.

I'd say we can just test without much more precision? Just ensure that the test ran for ~2 secs

Reply:
yes, I am truncating precision to 1 sec on both the duration and measured time